tensorflow/core/common_runtime/process_function_library_runtime.cc

```diff
@@ -637,6 +636,19 @@ Status ProcessFunctionLibraryRuntime::InstantiateMultiDevice(
   TF_RETURN_IF_ERROR(
       PartitionFunctionGraph(device_set, std::move(graph), &subgraphs));
 
+  for (const auto& pair : subgraphs) {
+    DumpGraph("Before running POST_PARTITIONING passes", pair.second.get());
+  }
+  optimization_options.graph = nullptr;
+  optimization_options.partition_graphs = &subgraphs;
+  TF_RETURN_IF_ERROR(OptimizationPassRegistry::Global()->RunGrouping(
```

Comment:
Add a comment like:

// Normally POST_PARTITIONING passes are run by distributed workers. Distributed workers are currently not supported in this code path, so we run the passes here.